integration-tests/src/test/java/oracle/weblogic/kubernetes/ItRetryImprovements.java

```diff
@@ -0,0 +1,298 @@
+        .append(FAILURE_RETRY_INTERVAL_SECONDS)
+        .append("\\s*seconds\\s*afterward\\s*until.*\\s*if\\s*the\\s*failure\\s*is\\s*not\\s*resolved").toString();
+
+    testUntil(() -> findStringInDomainMessage(retryOccurRegex), logger, "retry occurs as expected");
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    createSecretWithUsernamePassword(wlSecretName, domainNamespace, ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT);
+
+    // wait for the domain to exist
+    logger.info("Checking for domain custom resource in namespace {0}", domainNamespace);
+    testUntil(
+        domainExists(domainUid, DOMAIN_VERSION, domainNamespace),
+        logger,
+        "domain {0} to be created in namespace {1}",
+        domainUid,
+        domainNamespace);
+
+    // check that admin service/pod exists in the domain namespace
+    String adminServerPodName = domainUid + "-" + ADMIN_SERVER_NAME_BASE;
+    String managedServerPodNamePrefix = domainUid + "-" + MANAGED_SERVER_NAME_BASE;
+    logger.info("Checking that admin service/pod {0} exists in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace);
+
+    for (int i = 1; i <= replicaCount; i++) {
+      String managedServerPodName = managedServerPodNamePrefix + i;
+
+      // check that ms service/pod exists in the domain namespace
+      logger.info("Checking that clustered ms service/pod {0} exists in namespace {1}",
+          managedServerPodName, domainNamespace);
+      checkPodReadyAndServiceExists(managedServerPodName, domainUid, domainNamespace);
+    }
+  }
+
+  private boolean findStringInDomainMessage(String regex) {
+    // get the domain status message
+    StringBuffer getDomainInfoCmd = new StringBuffer("kubectl get domain/");
+    getDomainInfoCmd
+        .append(domainUid)
+        .append(" -n ")
+        .append(domainNamespace)
+        .append(" -o jsonpath='{.status.message}' --ignore-not-found");
+    logger.info("Command to get domain status message: " + getDomainInfoCmd);
+
+    CommandParams params = new CommandParams().defaults();
+    params.command(getDomainInfoCmd.toString());
+    ExecResult execResult = Command.withParams(params).executeAndReturnResult();
+    logger.info("Search: {0} in Domain status message: {1}", regex, execResult.stdout());
+
+    // match regex in domain info
+    Pattern pattern = Pattern.compile(regex);
+    Matcher matcher = pattern.matcher(execResult.stdout());
+
+    return matcher.find();
+  }
+
+  private boolean findStringInOperatorLog(String regex) {
+    // get operator pod name
+    String operatorPodName = assertDoesNotThrow(
+        () -> getOperatorPodName(OPERATOR_RELEASE_NAME, opNamespace));
+    assertNotNull(operatorPodName, "Operator pod name returned is null");
+    logger.info("Operator pod name {0}", operatorPodName);
+
+    // get the Operator logs
```

Comment:
We have the getPodLog() method in Kubernetes.java. You can reuse that.

Reply:
changed